src/com/lftechnology/java/training/sanish/generics/GenericTest.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+ */
+public class GenericTest {
+    private static final Logger LOGGER = Logger.getLogger(VehicleListImpl.class.getName());
+
+    private GenericTest() {
+
+    }
+
+    public static void main(String[] args) {
+        // add two number using generic function
+        addIntegerNumbers(2, 5);
+        addDoubleNumbers(3.5, 7.75);
+
+        // add / remove vehicle list of diffenrent type with generic class
+        workOnCarList();
+        workOnBikeList();
+
+        //Instantiate an object of Product using different data types.
+        Product computerProduct = new Product(new Computer("Dell", "Intel® Core™ i5-3337U CPU @ 1.80GHz × 4 ", "7.7 GiB", "500.0 GB"));
+        computerProduct.displayProductInfo();
+
+        Product mouseProduct = new Product(new Mouse("Delux", "Optical mouse"));
+        mouseProduct.displayProductInfo();
+
+        // display customers information using wildcard function
+        List<BankCustomer> bankCustomers = new ArrayList<BankCustomer>();
+        bankCustomers.add(new BankCustomer(1, "Sanish", "9841115968", 145258696, "Global", "Saving"));
+        displayList(bankCustomers);
+
+        List<RestaurantCustomer> restaurantCustomers = new ArrayList<RestaurantCustomer>();
+        restaurantCustomers.add(new RestaurantCustomer(1, "Sanish", "9841115968", "Cafe time"));
+        restaurantCustomers.add(new RestaurantCustomer(2, "Manish", "9841115968", "Thakali"));
+        displayList(restaurantCustomers);
+    }
+
+    /**
+     * Add two integer number using generic function
+     *
+     * @param num1 {@link Integer}
+     * @param num2 {@link Integer}
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public static void addIntegerNumbers(int num1, int num2) {
+        NumberOperator<Integer> intNumOperator = new NumberOperator<Integer>();
+        Number sumIntNumber = intNumOperator.addTwoNumbers(num1, num2);
+        LOGGER.log(Level.INFO, "Sum of {0} + {1} = {2}", new Object[] { num1, num2, sumIntNumber });
+    }
+
+    /**
+     * Add two double number using generic function
+     *
+     * @param num1 {@link Double}
+     * @param num2 {@link Double}
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public static void addDoubleNumbers(double num1, double num2) {
+        NumberOperator<Double> doubleNumOperator = new NumberOperator<Double>();
+        Number sumDoubleNumber = doubleNumOperator.addTwoNumbers(num1, num2);
+        LOGGER.log(Level.INFO, "Sum of {0} + {1} = {2}", new Object[] { num1, num2, sumDoubleNumber });
+    }
+
+    /**
+     * Add / remove cars form car list using generic class
+     *
+     * @author Sanish Maharjan <sanishmaharjan@lftechnology.com>
+     */
+    public static void workOnCarList() {
+        VehicleList<Car> carList = new VehicleListImpl<Car>();
+        carList.addVehicle(new Car("BMW", "Black"));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Black" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1192)